geomstats/geometry/discrete_surfaces.py

```diff
@@ -320,8 +343,567 @@ def surface_metric_matrices(self, point):
+
+        This method computes :math:`G_{b_1} = b_1.g_q^{-1} <dh_+, dk_+>`,
+        with notations taken from .. [HSKCB2022].
+
+        Parameters
+        ----------
+        ginvdga : array-like, shape=[n_faces, 2, 2]
+            Product of the inverse of the surface metric matrices
+            with their differential at a.
+        ginvdgb : array-like, shape=[n_faces, 2, 2]
+            Product of the inverse of the surface metric matrices
+            with their differential at b.
+        areas_bp : array-like, shape=[n_faces,]
+            Areas of the faces of the surface given by the base point.
+
+        Returns
+        -------
+        _ : array-like, shape=[...]
+            Term of order 0, and coefficient b1, of the inner-product.
+
+        References
+        ----------
+        .. [HSKCB2022] "Elastic shape analysis of surfaces with second-order
+            Sobolev metrics: a comprehensive numerical framework".
+            arXiv:2204.04238 [cs.CV], 25 Sep 2022.
+        """
+        return self.b1 * gs.sum(
+            gs.einsum("...bii->...b", ginvdga)
+            * gs.einsum("...bii->...b", ginvdgb)
+            * areas_bp,
+            axis=-1,
+        )
+
+    def _inner_product_c1(self, point_a, point_b, normals_bp, areas_bp):
+        r"""Compute c1 term of order 1 within the inner-product.
+
+        Denote h and k the tangent vectors a and b respectively.
+        Denote q the base point, i.e. the surface.
+
+        The equation of the inner-product is:
+        :math:`\int_M (G_{a_0} + G_{a_1} + G_{b_1} + G_{c_1} + G_{d_1} + G_{a_2})vol_q`.
+
+        This method computes :math:`G_{c_1} = c_1.g_q^{-1} <dh_\perp, dk_\perp>`,
+        with notations taken from .. [HSKCB2022].
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., n_vertices, 3]
+            Point a corresponding to tangent vec a.
+        point_b : array-like, shape=[..., n_vertices, 3]
+            Point b corresponding to tangent vec b.
+        normals_bp : array-like, shape=[n_faces, 3]
+            Normals of each face of the surface given by the base point.
+        areas_bp : array-like, shape=[n_faces,]
+            Areas of the faces of the surface given by the base point.
+
+        Returns
+        -------
+        _ : array-like, shape=[...]
+            Term of order 0, and coefficient c1, of the inner-product.
+
+        References
+        ----------
+        .. [HSKCB2022] "Elastic shape analysis of surfaces with second-order
+            Sobolev metrics: a comprehensive numerical framework".
+            arXiv:2204.04238 [cs.CV], 25 Sep 2022.
+        """
+        dna = self._space.normals(point_a) - normals_bp
+        dnb = self._space.normals(point_b) - normals_bp
+        return self.c1 * gs.sum(
+            gs.einsum("...bi,...bi->...b", dna, dnb) * areas_bp, axis=-1
+        )
+
+    def _inner_product_d1(
```

Comment:
we can very likely ignore the first axis in this function such that `...` is actually valid, i.e. from a quick reading of the function it seems we need  to pass e.g. `one_forms_a` as 4d (which is not what the docstrings suggest). I think we can handle easily 3d and 4d cases simultaneously.